### PR TITLE
Re organizing of the user LMMS directory

### LINKED ISSUE #1807: Splitting -- Placeholder folders (and files) in user directory for Ladspa, VSTs, themes and templates
This issue is to keep track of progress in #1541 as it covers a lot or areas. I am planing on breaking it down into the following areas

* [x] **Settings Dialog**
 * [x] Add Scrollbar to path selection list
 * [x] Move STK raw waves,  Default SF2, to bottom of list
 * [x] This dialog needs instructions and possibly enhanced to allow multiple selections. It's not obvious that you can have more than one search path for some of these values, that should be spelled out IMO.  
 * [x] Artwork should be renamed "Themes" in my opinion, kept on paths page.

* [x] **Folders**
 * [x] Sample, GIG, SF2, folders created (#1541)
 * [x] Add folders to setting tab
 * [x] Connect up to make these folders the default when open this type of file.    

* [x] **Plugins**
 * [x] Add user directories,  
 * [x] Options added to setting dialog

  
  
I plan on keeping this initial post updated

### PR DESCRIPTION
**Pull request by curlymorphic**
Have updated the users LMMS directory to contain the following folders

````
lmms
____projects
____templates
____samples
________gig
________sf2
____presets
____themes
____plugins
________ladspa
________vst
````

Added the extra folders to the settings dialog , Added a Scroll area to the dialog to allow the extra space needed.

Updated the icon where multiple Folders are allowed.
ArtWork renamed to themes

Opening of correct folders where needed, SF2 player opens the sf folder, GIG player opens the gig folder.

fixes #1807

### CODE REVIEW

File: src/core/ConfigManager.cpp [hunk 1]
```diff
@@ -425,6 +441,11 @@ void ConfigManager::loadConfigFile()
 		QDir().mkpath( userProjectsDir() );
 		QDir().mkpath( userSamplesDir() );
 		QDir().mkpath( userPresetsDir() );
+		QDir().mkpath( userGigDir() );
+		QDir().mkpath( userSf2Dir() );
+		QDir().mkpath( vstDir() );
```
Comment (tresf) [inline src/core/ConfigManager.cpp]: Noticed the naming convention of this `vstDir()` vs. `usersVstDir()`... also many Windows users historically have the `vstDir()` set to a restricted area (such as `Program Files/Steinberg/`).  Does `mkpath(...)` fail silently if the area can't be written to?

File: src/gui/SetupDialog.cpp [hunk 2]
```diff
@@ -394,37 +416,19 @@ SetupDialog::SetupDialog( ConfigTabs _tab_to_open ) :
...
 	TabWidget * artwork_tw = new TabWidget( tr(
-					"Artwork directory" ).toUpper(),
-								paths );
+					"Themes directory" ).toUpper(),
+								pathSelectors );
 	artwork_tw->setFixedHeight( 48 );
 
 	m_adLineEdit = new QLineEdit( m_artworkDir, artwork_tw );
-	m_adLineEdit->setGeometry( 10, 20, 300, 16 );
+	m_adLineEdit->setGeometry( 10, 20, txtLength, 16 );
```
Comment (tresf) [inline src/gui/SetupDialog.cpp]: I'm glad you broke the `300` geometry value out into a variable here... I assume we'll eventually want to do this with all of the repeated variables.

**Comment by tresf** [comment]
This isn't your fault, but we have some naming convention conflicts in our code, i.e. `users`**`Vst`**`Dir()` vs. `open`**`VST`**`Dir()`.  This may be the time to re-factor this. What do you think?

-Tres

**Comment by tresf** [comment]
@curlymorphic this is great.  Since it is a significant change, I want a few set of eyes before merging it.  @badosu, @Lukas-W any objections?

File: src/core/ConfigManager.cpp (hunk 1 above)
Comment (curlymorphic) [inline src/core/ConfigManager.cpp]: The directories are only created once, so there will be a user Lmms/plugin/vst folder created when the the user lmms folder in initially created.

If/ when the user sets it to another location ie ```` Program Files/Steinberg/```` no attempt is made to create or write to that location.

Comment (tresf) [inline src/core/ConfigManager.cpp]: > If/ when the user sets it to another location ie Program Files/Steinberg/ no attempt is made to create or write to that location.

Ok. :)

File: src/gui/SetupDialog.cpp (hunk 2 above)
Comment (curlymorphic) [inline src/gui/SetupDialog.cpp]: > I assume we'll eventually want to do this with all of the repeated variables.

I feel this should be done as a separate pull request. mainly to keep this pull request to functionality. Also to keep the code review more manageable.

Comment (tresf) [inline src/gui/SetupDialog.cpp]: > I feel this should be done as a separate pull request.

:+1:

**Comment by curlymorphic** [comment]
> This isn't your fault, but we have some naming convention conflicts in our code, i.e. usersVstDir() vs. openVSTDir(). This may be the time to re-factor this. What do you think?

will have a look at this later today

**Comment by curlymorphic** [comment]
> This isn't your fault, but we have some naming convention conflicts in our code, i.e. usersVstDir() vs. openVSTDir(). This may be the time to re-factor this. What do you think?

Added ````userVstDir()```` to configManager.h for consistancy.

**Comment by Spekular** [comment]
Wait, where do "actual" (wav) samples go in this? It seems they go under
samples, but we also have subfolders there. This looks like it could get
pretty messy to me. Maybe there ought to be a subfolder specifially for
audio samples as well? Sorry if I'm late to the discussion.

**Comment by tresf** [comment]
> where do "actual" (wav) samples go in this? It seems they go under
samples

Yes, samples.

**Comment by tresf** [comment]
> Maybe there ought to be a subfolder specifially for
audio samples as well? Sorry if I'm late to the discussion.

Most people already put them in subfolders, you may be overthinking this..

**Comment by Spekular** [comment]
@tresf I know most people put their samples in subfolders, so do I. It just
feels unorganized to me to have multiple sample pack folders and the gig
and sf2 folders together.

**Comment by tresf** [comment]
Then don't.  What do you propose?  Samples/samples?

**Comment by tresf** [comment]
I think too many root folders is much more confusing than a few subfolders.

**Comment by curlymorphic** [comment]
> I think too many root folders is much more confusing than a few subfolders.

I agree, we have to be careful to still give users there freedom.

**Comment by Spekular** [comment]
Samples/audio? I would have proposed:
Song
_Projects
_Templates
Sound
_Gig
_Sf
_Samples
_Ladspa
_VST
___Instruments
___Effects
Themes

**Comment by Sti2nd** [comment]
Nah, leave it. SF2 and GIG was decided as samples. Samples is audio, no difference :)

**Comment by tresf** [comment]
Merging per organization feedback and researching #1807.  :+1: 

If we encounter issues, we'll open new bugs. :)

**Comment by tresf** [comment]
Oh, and updated release notes. :)
